src/patternfly/components/Button/button.scss

```diff
@@ -611,6 +622,12 @@
       }
     }
   }
+
+  &.pf-m-RTL-icon-static {
+    .#{$button}__icon {
+      @include pf-v5-mirror-inline-on-rtl
+    }
+  }
```

Comment:
This works but so far we tried not to have RTL modifiers for everything that accepts an icon, and instead use either the mixins in core or react has been using `<Icon shouldMirrorRTL />`. On the react side I'd expect this to be setup as

```
<Button variant="plain">
  <Icon shouldMirrorRTL><AngleRightIcon /></Icon>
</Button>
```

I'd probably do the same on the core side. You should always be able to pass the icon component to a button as the `icon` prop or as a child of a plain button. That also prevents any confusion around the global class `.pf-m-mirror-inline-rtl` and this local `.pf-m-RTL-icon-static` class, which are named differently, but do the same thing.